diff --git a/CMakeLists.txt b/CMakeLists.txt
index d0f0ec4..06c369b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -125,15 +125,15 @@ endif()

 if(JSON_VALIDATOR_INSTALL)
     # Set the install path to the cmake config files (Relative, so install works correctly under Hunter as well)
-    set(INSTALL_CMAKE_DIR "lib/cmake/${PROJECT_NAME}")
-    set(INSTALL_CMAKEDIR_ROOT share/cmake)
+    include(CMakePackageConfigHelpers)
+    set(INSTALL_CMAKE_DIR ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME})
+    set(INSTALL_CMAKEDIR_ROOT ${CMAKE_INSTALL_LIBDIR}/share/cmake/${PROJECT_NAME})

     # Install Targets
     install(EXPORT ${PROJECT_NAME}Targets
             FILE ${PROJECT_NAME}Targets.cmake
             DESTINATION "${INSTALL_CMAKE_DIR}")

-    include(CMakePackageConfigHelpers)
     write_basic_package_version_file(
         ${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}ConfigVersion.cmake
         VERSION ${PROJECT_VERSION}
@@ -141,9 +141,9 @@ if(JSON_VALIDATOR_INSTALL)
         )

     configure_package_config_file(
-        ${PROJECT_SOURCE_DIR}/${PROJECT_NAME}Config.cmake.in
+        ${CMAKE_CURRENT_SOURCE_DIR}/${PROJECT_NAME}Config.cmake.in
         ${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}Config.cmake
-        INSTALL_DESTINATION ${INSTALL_CMAKEDIR_ROOT}/${PROJECT_NAME}
+        INSTALL_DESTINATION ${INSTALL_CMAKE_DIR}
         )

     install(
@@ -153,4 +153,4 @@ if(JSON_VALIDATOR_INSTALL)
         DESTINATION
             ${INSTALL_CMAKE_DIR}
         )
-endif()
\ No newline at end of file
+endif()
